lib/backend/syncersv1/updateprocessors/rules.go

```diff
@@ -154,6 +203,47 @@ func parseNamespaceSelector(s string) string {
 	return updated
 }
 
+// parseServiceAccounts takes a v2 service account selector and returns the appropriate v1 representation
+// by prefixing the keys with the `pcsa/` prefix. For example, `k == 'v'` becomes `pcsa/k == 'v'`.
+func parseServiceAccounts(sam *apiv3.ServiceAccountMatch) string {
+
+	parsedSelector, err := parser.Parse(sam.Selector)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to parse service account selector: %s", sam.Names)
+		return ""
+	}
+
+	parsedSelector.AcceptVisitor(parser.PrefixVisitor{Prefix: conversion.ServiceAccountLabelPrefix})
+	updated := parsedSelector.String()
+	log.WithFields(log.Fields{"original": sam.Selector, "updated": updated}).Debug("Updated service account selector")
+	if len(sam.Names) == 0 {
+		return updated
+	}
+
+	// Convert the list of ServiceAccounts to selector
+	var namesSelector, comma string
```

Comment:
Might be cleaner to do something like this: https://play.golang.org/p/K9L10y2dLp

Not sure if it's just me, but seems more digestible to me and is fewer lines of code.